pkg/lib/oauth/handler/handler_authz.go

```diff
@@ -534,6 +570,7 @@ func (h *AuthorizationHandler) validateRequest(
 	if len(allowedResponseTypes) == 0 {
 		allowedResponseTypes = []string{"code"}
 	}
+	allowedResponseTypes = append(allowedResponseTypes, whiteslistedResponseTypes...)
```

Comment:
line 570 to 572 is redundant now, because we always allow whitelistedResponseTypes. Remove those lines.